ibis/util.py

```diff
@@ -30,6 +32,35 @@
 V = TypeVar("V")
 
 
+class frozendict(Mapping, Hashable):
+
+    __slots__ = ("_dict", "_hash")
+
+    def __init__(self, *args, **kwargs):
+        # create a read-only view of the underlying mutable mapping
+        # self._dict = types.MappingProxyType(dict(*args, **kwargs))
```

Comment:
Remove the `types.MappingProxyType` since it prevent the object from being pickled.